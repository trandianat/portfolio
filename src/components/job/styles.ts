import { css } from '@emotion/react';

export const job = css({
  display: 'flex',
  flexDirection: 'column',
  h2: { fontFamily: 'var(--serif-italic)' },
  ul: { marginTop: 6 },
});
