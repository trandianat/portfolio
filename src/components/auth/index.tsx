import { useAuthenticator, withAuthenticator } from '@aws-amplify/ui-react';
import { Fragment, ReactNode } from 'react';
import { Link, useLocation } from 'react-router-dom';
import * as styles from 'components/auth/styles';
import isProjects from 'utils/projects';

const Auth = ({ children }: { children: ReactNode }): JSX.Element => {
  const { signOut } = useAuthenticator();
  const { pathname } = useLocation();
  return (
    <Fragment>
      <div css={styles.navigation}>
        {isProjects(pathname) ? <div /> : <Link to="projects">â‰ª Back</Link>}
        <button onClick={signOut}>Sign out</button>
      </div>
      {children}
    </Fragment>
  );
};

export default withAuthenticator(Auth, {
  components: { SignIn: { Footer: () => <Fragment /> } },
  formFields: {
    signIn: {
      username: { label: '', placeholder: 'Username' },
      password: { label: '', placeholder: 'Password' },
    },
  },
  hideSignUp: true,
  // services: {
  //   handleSignIn: async ({
  //     username,
  //     password,
  //   }: {
  //     username: string;
  //     password: string;
  //   }) => console.log('Credentials: ' + username + ' / ' + password),
  // },
});
