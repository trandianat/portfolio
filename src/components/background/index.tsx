import { PropsWithChildren } from 'react';
import * as styles from 'components/background/styles';
import { Color } from 'utils/types';

export const Background = ({
  children,
  color = Color.WHITE,
}: PropsWithChildren<{ color?: Color }>): JSX.Element => {
  return (
    <div css={styles.background(color)}>
      <div css={styles.container}>{children}</div>
    </div>
  );
};
