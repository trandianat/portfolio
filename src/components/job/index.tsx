import * as styles from 'components/job/styles';

export const Job = ({
  role,
  company,
  start,
  end,
  skills,
  scope,
  accomplishments,
  indent = false,
}: {
  role: string;
  company: string;
  start: string;
  end: string;
  skills: string[];
  scope: string;
  accomplishments: string[];
  indent?: boolean;
}): JSX.Element => (
  <div css={styles.job(indent)}>
    <div css={styles.verticalDates}>
      <div css={styles.verticalRange}>
        <div className="sans">{end}</div>
        <div className="sans">{start}</div>
      </div>
      <div css={styles.verticalTimeline}>
        <div css={styles.circle} />
        <div css={styles.verticalLine} />
        <div css={styles.circle} />
      </div>
    </div>
    <div css={styles.horizontalDates}>
      <div css={styles.horizontalTimeline}>
        <div css={styles.circle} />
        <div css={styles.horizontalLine} />
        <div css={styles.circle} />
      </div>
      <div css={styles.horizontalRange}>
        <div className="sans">{start}</div>
        <div className="sans">{end}</div>
      </div>
    </div>
    <div css={styles.role}>
      <h2>{role}</h2>
      <p className="sans">{company}</p>
      <div css={styles.skills}>
        {skills.map((skill: string) => (
          <div css={styles.skill} key={skill}>
            {skill}
          </div>
        ))}
      </div>
      <p>{scope}</p>
      <ul>
        {accomplishments.map((accomplishment: string) => (
          <li key={accomplishment}>{accomplishment}</li>
        ))}
      </ul>
    </div>
  </div>
);
