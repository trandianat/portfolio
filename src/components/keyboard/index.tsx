import { useEffect, useState } from 'react';
import * as styles from 'components/keyboard/styles';
import { Keyboard as Key } from 'utils/constants';

export const Keyboard = ({
  disabled,
  guesses,
  handleSubmit,
  input,
  setInput,
}: {
  disabled: boolean;
  guesses: string[];
  handleSubmit: any;
  input: string;
  setInput: any;
}): JSX.Element => {
  const [rows, setRows] = useState<Array<string[]>>([]);

  const isAction = (key: string): boolean =>
    [Key.DELETE, Key.ENTER].includes(key as Key);
  const isDelete = (key: string): boolean => key === Key.DELETE;
  const isEnter = (key: string): boolean => key === Key.ENTER;

  const setLetterClass = (key: string, rowIndex: number): string => {
    let className = 'letter';
    if (isAction(key)) {
      className += ' action';
      if (rowIndex < rows.length - 1) {
        className += ' primary';
      }
    } else {
      const used = guesses.map(guess => [...guess]).flat();
      if (used.includes(key)) {
        className += ' used';
      }
    }
    return className;
  };

  useEffect(() => {
    const row1 = ['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'];
    const row2 = ['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'];
    const row3 = [Key.ENTER, 'Z', 'X', 'C', 'V', 'B', 'N', 'M', Key.DELETE];
    const row4 = [Key.ENTER, Key.DELETE];
    setRows([row1, row2, row3, row4]);
  }, []);

  return (
    <div css={styles.keyboard}>
      {rows.map((row, index) => (
        <div className={`row row-${index + 1}`} key={`row-${index + 1}`}>
          {row.map((key: string) => (
            <button
              className={setLetterClass(key, index)}
              key={key}
              onClick={() => {
                if (isDelete(key)) {
                  setInput((input: string) => input.slice(0, -1));
                } else if (isEnter(key)) {
                  handleSubmit();
                } else {
                  setInput((input: string) => input + key);
                }
              }}
              {...(((isDelete(key) && !input.length) ||
                (isEnter(key) && input.length < 5) ||
                (!isAction(key) && input.length === 5) ||
                disabled) && {
                disabled: true,
              })}
            >
              {key}
            </button>
          ))}
        </div>
      ))}
    </div>
  );
};
