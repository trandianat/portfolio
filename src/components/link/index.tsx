import { PropsWithChildren, ReactNode } from 'react';
import * as styles from 'components/link/styles';
import { Position, Size } from 'utils/constants';

export const Link = ({
  children,
  icon,
  iconGap = Size.SMALL,
  iconPosition = Position.RIGHT,
  text,
  url,
  ...rest
}: PropsWithChildren<{
  icon?: ReactNode;
  iconGap?: Size;
  iconPosition?: Position;
  text: string;
  url: string;
  [rest: string]: unknown;
}>): JSX.Element => (
  <a
    css={styles.link(iconGap, iconPosition)}
    href={url}
    target="_blank"
    {...rest}
  >
    {iconPosition === Position.LEFT && <span className="icon">{icon}</span>}
    <span className="text">{text}</span>
    {iconPosition === Position.RIGHT && <span className="icon">{icon}</span>}
  </a>
);
