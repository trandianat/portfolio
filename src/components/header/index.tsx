import { ReactNode } from 'react';
import { Background } from 'components/background';
import * as styles from 'components/header/styles';

export const Header = ({ children }: { children: ReactNode }): JSX.Element => (
  <Background>
    <div css={styles.header}>
      <div css={styles.divider} />
      <div css={styles.name}>
        <div>
          <h1>Diana Tran</h1>
          <h2 className="sans">Software Engineer & Manager</h2>
        </div>
        <div css={styles.logo}>
          <div className="content">
            <div className="dot" />
            <div>DT</div>
            <div className="line" />
          </div>
        </div>
      </div>
      {children}
    </div>
  </Background>
);
