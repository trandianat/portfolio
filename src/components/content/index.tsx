import { Fragment, ReactNode } from 'react';
import { Background } from 'components/background';
import { Header } from 'components/header';
import { Navigation } from 'components/navigation';
import { Color } from 'utils/constants';

export const Content = ({ children }: { children: ReactNode }): JSX.Element => (
  <Fragment>
    <Header />
    <Background color={Color.GRAY}>
      <Navigation />
      {children}
    </Background>
  </Fragment>
);
