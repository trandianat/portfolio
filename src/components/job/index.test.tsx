import { create } from 'react-test-renderer';
import { Job } from 'components/job';

describe('Job', () => {
  it('should match the snapshot', () => {
    expect(
      create(
        <Job
          role="Role"
          company="Company"
          start="Start"
          end="End"
          skills={['Skill 1', 'Skill 2']}
          scope="Scope"
          accomplishments={['Accomplishment 1', 'Accomplishment 2']}
        />
      )
    ).toMatchSnapshot();
  });
});
