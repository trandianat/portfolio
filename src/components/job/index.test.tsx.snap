// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job should match the snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 16px;
}

@media (max-width: 450px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    gap: 2px;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 12px;
}

@media (max-width: 450px) {
  .emotion-1 {
    display: none;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 500;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  min-width: 70px;
  padding: 8px 0;
  text-align: right;
  white-space: nowrap;
}

.emotion-3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 8px 0;
}

.emotion-4 {
  background-color: var(--white);
  border: 2px solid var(--dark-gray);
  border-radius: 8px;
  min-height: 8px;
  min-width: 8px;
}

.emotion-5 {
  background-color: var(--dark-gray);
  height: 100%;
  width: 2px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 500;
  gap: 2px;
}

@media (min-width: 451px) {
  .emotion-7 {
    display: none;
  }
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-10 {
  background-color: var(--dark-gray);
  height: 2px;
  width: 100%;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 6px;
  margin: 6px 0;
}

.emotion-15 {
  background-color: var(--black);
  border-radius: 12px;
  color: var(--white);
  font: .685rem var(--sans);
  letter-spacing: 1px;
  padding: 2px 8px;
  text-transform: uppercase;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <div
        className="sans"
      >
        End
      </div>
      <div
        className="sans"
      >
        Start
      </div>
    </div>
    <div
      className="emotion-3"
    >
      <div
        className="emotion-4"
      />
      <div
        className="emotion-5"
      />
      <div
        className="emotion-4"
      />
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-8"
    >
      <div
        className="emotion-4"
      />
      <div
        className="emotion-10"
      />
      <div
        className="emotion-4"
      />
    </div>
    <div
      className="emotion-12"
    >
      <div
        className="sans"
      >
        Start
      </div>
      <div
        className="sans"
      >
        End
      </div>
    </div>
  </div>
  <div
    className="emotion-13"
  >
    <h2>
      Role
    </h2>
    <p
      className="sans"
    >
      Company
    </p>
    <div
      className="emotion-14"
    >
      <div
        className="emotion-15"
      >
        Skill 1
      </div>
      <div
        className="emotion-15"
      >
        Skill 2
      </div>
    </div>
    <p>
      Scope
    </p>
    <ul>
      <li>
        Accomplishment 1
      </li>
      <li>
        Accomplishment 2
      </li>
    </ul>
  </div>
</div>
`;
